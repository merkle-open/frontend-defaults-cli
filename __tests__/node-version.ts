import { defaultIt } from './shared';

describe('node-version', async () => {
	it('default', async () => await defaultIt('node-version', '--nodeVersion --force --noInstall'));
	it('with-githooks', async () => await defaultIt('node-version-with-githooks', '--nodeVersion --githooks --force --noInstall'));
});
