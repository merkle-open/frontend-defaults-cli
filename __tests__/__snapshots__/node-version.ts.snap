// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node-version default 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "0.1.0-mock",
    }
  }
/>
`;

exports[`node-version default 2`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "0.1.0-mock",
    }
  }
/>
`;

exports[`node-version default 3`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "{
  \\"engines\\": {
    \\"node\\": \\">=0\\"
  }
}",
    }
  }
/>
`;

exports[`node-version default 4`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": ".node-version,.nvmrc,package.json",
    }
  }
/>
`;

exports[`node-version with-githooks 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "# ~/.huskyrc
# Note: it will run in /bin/sh shell
if command -v nodenv >/dev/null 2>&1; then
  eval \\"$(nodenv init -)\\"
fi",
    }
  }
/>
`;

exports[`node-version with-githooks 2`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "0.1.0-mock",
    }
  }
/>
`;

exports[`node-version with-githooks 3`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "0.1.0-mock",
    }
  }
/>
`;

exports[`node-version with-githooks 4`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "{
  \\"engines\\": {
    \\"node\\": \\">=0\\"
  }
}",
    }
  }
/>
`;

exports[`node-version with-githooks 5`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": ".huskyrc,.node-version,.nvmrc,package.json",
    }
  }
/>
`;
